api-tests/src/steps/cri-steps.ts

```diff
@@ -48,12 +50,18 @@ When(
       throw new Error("Last journey engine response was not a CRI response");
     }
 
+    const matchedRegex = /^(.*)-with-expiry$/.exec(scenario);
+    const nbf = matchedRegex === null ? undefined : EXPIRED_NBF;
+    const extractedScenario =
+      matchedRegex === null ? scenario : matchedRegex[1];
```

Comment:
Wanted to re-use the stub data we already had so used regex to extract them but can get rid of this and just duplicate the data instead if this isn't as readable

Reply:
This is ok, but if you use a cucumber step for this, you might get better autocomplete and it'll read nicer...

The simplest approach is to pull out the action as a method and define two steps that use it:
```
const submitCriDetails = async (world: World, scenario: string, nbf?: number): Promise<void> => {
  
};

When(
  "I submit {string} details to the CRI stub",
  async function (this: World, scenario: string): Promise<void> {
    await submitCriDetails(this, scenario);
  },
);

When(
  "I submit expired {string} details to the CRI stub",
  async function (this: World, scenario: string): Promise<void> {
    await submitCriDetails(this, scenario, EXPIRED_NBF);
  },
);
```

But a cleverer approach might be to switch to a regex step:
```
When(
  /I submit (expired )?'([\w-]+)' details to the CRI stub/,
  async function (this: World, expired: string, scenario: string): Promise<void> {
    await submitCriDetails(this, scenario, expired ? EXPIRED_NBF : undefined);
  },
);
```
(`expired` will have the value `expired ` if included, or undefined)

Reply:
I tried using a regex too but wasn't as nice as this thanks! :)